src/bindings/c/include/openvino/c/ov_infer_request.h

```diff
@@ -70,37 +71,204 @@ OPENVINO_C_API(ov_status_e)
+ * @ingroup infer_request
+ * @param infer_request A pointer to the ov_infer_request_t.
+ * @param port Port of the input or output tensor, which can be got by call interface from ov_model_t/ov_compiled_model_t.
+ * @param tensor Reference to the tensor.
+ * @return Status code of the operation: OK(0) for success.
+ */
+OPENVINO_C_API(ov_status_e)
+ov_infer_request_set_tensor_by_port(ov_infer_request_t* infer_request, const ov_output_node_t* port, const ov_tensor_t* tensor);
+
+/**
+ * @brief Sets an input/output tensor to infer on by the const port of input/output tensor.
+ * @ingroup infer_request
+ * @param infer_request A pointer to the ov_infer_request_t.
+ * @param port Const port of the input or output tensor, which can be got by call interface from ov_model_t/ov_compiled_model_t.
+ * @param tensor Reference to the tensor.
+ * @return Status code of the operation: OK(0) for success.
+ */
+OPENVINO_C_API(ov_status_e)
+ov_infer_request_set_tensor_by_const_port(ov_infer_request_t* infer_request, const ov_output_const_node_t* port, const ov_tensor_t* tensor);
+
+/**
+ * @brief Sets a batch of tensors for input data to infer by tensor name.
+ * Model input must have batch dimension, and the number of @p tensors must match the batch size.
+ * The current version supports setting tensors to model inputs only. If @p tensor_name is associated
+ * with output (or any other non-input node), an exception is thrown.
+ * @ingroup infer_request
+ * @param infer_request A pointer to the ov_infer_request_t.
+ * @param tensor_name Name of the input tensor.
+ * @param tensors Input tensors for batched infer request. The type of each tensor must match the model
+ * input element type and shape (except batch dimension). Total size of tensors must match the input size.
+ * @return Status code of the operation: OK(0) for success.
+ */
+OPENVINO_C_API(ov_status_e)
+ov_infer_request_set_tensors(ov_infer_request_t* infer_request, const char* tensor_name, const ov_tensor_list_t* tensors);
+
+/**
+ * @brief Sets a batch of tensors for input data to infer by input port.
+ * Model input must have batch dimension, and the number of @p tensors must match the batch size.
+ * The current version supports setting tensors to model inputs only. If @p port is associated
+ * with output (or any other non-input node), an exception is thrown.
+ * @ingroup infer_request
+ * @param infer_request A pointer to the ov_infer_request_t.
+ * @param port Const port of the input tensor.
+ * @param tensors Input tensors for batched infer request. The type of each tensor must match the model
+ * input element type and shape (except batch dimension). Total size of tensors must match the input size.
+ * @return Status code of the operation: OK(0) for success.
+ */
+OPENVINO_C_API(ov_status_e)
+ov_infer_request_set_tensors_by_const_port(ov_infer_request_t* infer_request, const ov_output_const_node_t* port, const ov_tensor_list_t* tensors);
```

Comment:
The same

Reply:
As we aligned, C API will not support set tensor list for infer_request.